test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -0,0 +1,342 @@
+    private static final String SIZE_30_EVENT = "data of size 30";
+
+    private static final int READ_TIMEOUT = 1000;
+    private static final int MAX_SIZE_IN_STREAM = 180;
+    private static final int MIN_SIZE_IN_STREAM = 90;
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(7 * 60);
+
+    private final ReaderConfig readerConfig = ReaderConfig.builder().build();
+    private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(4, "executor");
+    private final ScheduledExecutorService streamCutExecutor = ExecutorServiceHelpers.newScheduledThreadPool(2, "streamCutExecutor");
+    private URI controllerURI = null;
+    private StreamManager streamManager = null;
+    private Controller controller = null;
+
+    /**
+     * This is used to setup the various services required by the system test framework.
+     *
+     * @throws MarathonException    when error in setup
+     */
+    @Environment
+    public static void initialize() throws MarathonException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+
+        controller = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(clientConfig)
+                .maxBackoffMillis(5000).build(), executor);
+        streamManager = StreamManager.create(clientConfig);
+
+        assertTrue("Creating scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream", streamManager.createStream(SCOPE, STREAM,
+                StreamConfiguration.builder()
+                        .scalingPolicy(ScalingPolicy.fixed(1))
+                        .retentionPolicy(RetentionPolicy.bySizeBytes(MIN_SIZE_IN_STREAM, MAX_SIZE_IN_STREAM)).build()));
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
+        ExecutorServiceHelpers.shutdown(streamCutExecutor);
+    }
+
+    @Test
+    public void multipleSubscriberCBRTest() throws Exception {
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(SCOPE, clientConfig);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+
+        // Write events.
+        log.info("ConsumptionBasedRetentionWithMultipleReaderGroupsTest -> Writing event e1 to {}/{}", SCOPE, STREAM);
```

Comment:
NIT : Log can be updated as log.info("Writing event e1 to {}/{}", SCOPE, STREAM);
This is applicable in every log line.

Reply:
done.